src/main/java/org/zalando/nakadi/repository/kafka/KafkaCursor.java

```diff
@@ -24,63 +29,78 @@ public long getOffset() {
+    public KafkaCursor addOffset(final long toAdd) {
+        return new KafkaCursor(topic, partition, offset + toAdd);
     }
 
-    public static KafkaCursor fromNakadiCursor(final Cursor cursor) {
-        return kafkaCursor(cursor.getPartition(), cursor.getOffset());
+    public NakadiCursor toNakadiPosition() {
+        return new NakadiCursor(topic, toNakadiPartition(partition), toNakadiOffset(offset));
     }
 
-    public static KafkaCursor kafkaCursor(final int partition, final long offset) {
-        return new KafkaCursor(partition, offset);
+    public static String toNakadiOffset(final long offset) {
+        return String.valueOf(offset);
     }
 
-    public static KafkaCursor kafkaCursor(final String partition, final String offset) {
-        return new KafkaCursor(toKafkaPartition(partition), toKafkaOffset(offset));
+    public static String toNakadiPartition(final int partition) {
+        return String.valueOf(partition);
     }
 
     public static int toKafkaPartition(final String partition) {
         return Integer.parseInt(partition);
     }
 
-    public static String toNakadiPartition(final int partition) {
-        return Integer.toString(partition);
+    public static KafkaCursor fromNakadiPosition(final NakadiCursor tp) throws InvalidCursorException {
+        final Integer partition;
+        try {
+            partition = toKafkaPartition(tp.getPartition());
+        } catch (final NumberFormatException ex) {
+            throw new InvalidCursorException(PARTITION_NOT_FOUND, tp);
+        }
+        try {
+            return new KafkaCursor(tp.getTopic(), partition, toKafkaOffset(tp.getOffset()));
+        } catch (final NumberFormatException ex) {
+            throw new InvalidCursorException(INVALID_FORMAT, tp);
+        }
     }
 
-    public static long toKafkaOffset(final String offest) {
-        return Long.parseLong(offest);
+    public static long toKafkaOffset(final String offset) {
+        return Long.parseLong(offset);
     }
 
-    public static String toNakadiOffset(final long offset) {
-        return Long.toString(offset);
+    @Override
+    public int compareTo(final KafkaCursor other) {
+        return Long.compare(getOffset(), other.getOffset());
     }
 
     @Override
     public boolean equals(final Object o) {
         if (this == o) {
             return true;
         }
-        if (o == null || getClass() != o.getClass()) {
+        if (!(o instanceof KafkaCursor)) {
             return false;
         }
 
         final KafkaCursor that = (KafkaCursor) o;
 
-        return partition == that.partition && offset == that.offset;
+        return Objects.equals(topic, that.topic)
+                && partition == that.partition
+                && offset == that.offset;
     }
 
     @Override
     public int hashCode() {
-        int result = partition;
+        int result = topic != null ? topic.hashCode() : 0;
```

Comment:
We already had conversation about nulls in equals/hashCode/toString(), I do not want to start it again